spec/system/provider_interface/reject_an_application_spec.rb

```diff
@@ -0,0 +1,151 @@
+
+    click_on 'Continue'
+  end
+
+  def and_i_check_the_reasons_for_rejection
+    expect(page).to have_content('Training provider feedback')
+    expect(page).to have_content('Here’s why your application was unsuccessful')
+
+    expect(page).to have_content('Something you did')
+    expect(page).to have_content("There was no need to sing 'Run to the Hills' for us\nAdvice: Leave the singing out next time")
+
+    expect(page).to have_content('Quality of application')
+    expect(page).to have_content('Telling people you are a stable genius might be a bit loaded')
+
+    expect(page).to have_content('Qualifications')
+    expect(page).to have_content('No Maths GCSE grade 4 (C) or above, or valid equivalent')
+    expect(page).to have_content('No degree')
+
+    expect(page).to have_content('Performance at interview')
+    expect(page).to have_content("Don't sing 'Run to the Hills' at the start of the interview")
+
+    expect(page).to have_content('Honesty and professionalism')
+    expect(page).to have_content('We cannot accept references from your mum')
+
+    expect(page).to have_content('Safeguarding issues')
+    expect(page).to have_content('You abducted Jenny, now Matrix is coming to find her')
+  end
+
+  def and_i_choose_to_change_some_reasons_for_rejection
+    click_on 'Change', match: :first
+
+    expect(page).to have_checked_field 'provider-interface-reasons-for-rejection-safeguarding-y-n-yes-field'
+    expect(page).to have_checked_field 'provider-interface-reasons-for-rejection-honesty-and-professionalism-y-n-yes-field'
+
+    choose 'provider-interface-reasons-for-rejection-honesty-and-professionalism-y-n-no-field'
+    choose 'provider-interface-reasons-for-rejection-safeguarding-y-n-no-field'
+
+    click_on 'Continue'
+  end
+
+  def and_i_answer_additional_reasons
+    choose 'provider-interface-reasons-for-rejection-other-advice-or-feedback-y-n-yes-field'
+    fill_in 'provider-interface-reasons-for-rejection-other-advice-or-feedback-details-field', with: 'While impressive, your parkour skills are not relevant'
+
+    choose 'provider-interface-reasons-for-rejection-interested-in-future-applications-y-n-yes-field'
+
+    click_on 'Continue'
+  end
+
+  def and_i_check_the_amended_reasons_for_rejection
+    expect(page).to have_content('The provider would be interested in future applications from you')
+
+    expect(page).not_to have_content('Honesty and professionalism')
+    expect(page).not_to have_content('Safeguarding issues')
+
+    expect(page).to have_content('Additional advice')
+    expect(page).to have_content('While impressive, your parkour skills are not relevant')
+  end
+
+  def and_i_submit_the_reasons_for_rejection
+    click_on 'Reject application'
+  end
+
+  def then_i_can_see_the_reasons_were_saved
+    expect(page).to have_content('Application rejected')
```

Comment:
I imagine this step will include some more checks around the display of structured rejection reasons?

Reply:
@mnacos see https://github.com/DFE-Digital/apply-for-teacher-training/pull/3404/commits/4ca7143c010b08116efbc151bfd5e56083a35c45 for component to display reasons for rejection

Reply:
@adamsilver https://github.com/DFE-Digital/apply-for-teacher-training/commit/27396ca292b495ce36c024e37b4c28464d64a0a7 should fix the inset text problem when displaying reasons for rejection.